src/backend/mailbox_manager/models.py

```diff
@@ -104,6 +111,61 @@ def is_identity_provider_ready(self) -> bool:
             bool(self.organization) and self.status == MailDomainStatusChoices.ENABLED
         )
 
+    def notify_status_change(self, recipients=None, language=None):
+        """
+        Notify the support team that the domain status has changed.
+        """
+        status_notification_mails = {
+            # new status domain: (mail subject, mail template html, mail template text)
+            MailDomainStatusChoices.ENABLED: (
+                _("[La Suite] Your domain is ready"),
+                "mail/html/maildomain_enabled.html",
+                "mail/text/maildomain_enabled.txt",
+            ),
+            MailDomainStatusChoices.ACTION_REQUIRED: (
+                _("[La Suite] Your domain requires action"),
+                "mail/html/maildomain_action_required.html",
+                "mail/text/maildomain_action_required.txt",
+            ),
+            MailDomainStatusChoices.FAILED: (
+                _("[La Suite] Your domain has failed"),
+                "mail/html/maildomain_failed.html",
+                "mail/text/maildomain_failed.txt",
+            ),
+        }
```

Comment:
Note: This constant should not be instantiated at every function call IMHO, this can be put outside of the class/method.